pages/api/retrieval.ts

```diff
@@ -0,0 +1,171 @@
+          Key: {
+            SubmissionID: {
+              S: submissionID,
+            },
+            FormID: {
+              S: formID,
+            },
+          },
+          UpdateExpression: "SET Retrieved = :retrieved",
+          ConditionExpression: "SubmissionID = :submission",
+          ExpressionAttributeValues: {
+            ":retrieved": { N: "1" },
+            ":submission": { S: submissionID },
+          },
+          ReturnValues: "NONE",
+        };
+        //Update one item at the time
+        await db.send(new UpdateItemCommand(updateItem));
+      }
+    }
+    //Return responses data
+    return res.status(200).json({ responses: formResponses.Items });
+  } catch (error) {
+    logMessage.warn(`Possible data loss for these submissionIDs`);
+    //print submission ids as string
+    logMessage.warn(submissionIDlist ? submissionIDlist.toString() : "Empty submissionID list");
+    logMessage.error(error);
+    return res.status(500).json({ responses: "Error on Server Side" });
+  }
+}
+
+/**
+ * @description
+ * This function will make sure that the incoming request is valid by enforcing a set of
+ * rules.
+ * Request parameters :
+ *  - The maxRecords exists and it ranges between 1 and 10.
+ *  - A formID exists and is valid i.e 78
+ * BearerToken :
+ *  - Must contains an email address
+ *  - It has not expired
+ *  - It exists in our Database
+ *  - The record found in the database includes the email, active, and template_id fields
+ *  - The email on the token record matches the email in the payload.
+ *  - The active flag boolean is true.
+ * @param handler - A method to retrieve form's responses
+ * @returns
+ */
+export const formResponsesReqValidator = (
+  handler: (res: NextApiResponse, formID: string, maxRecords: number) => void
+) => {
+  return async function (req: NextApiRequest, res: NextApiResponse): Promise<unknown> {
+    try {
+      //Default value to 10 if it's undefined
+      const { maxRecords = 10, formID } = req.query;
+      //Get formID form the baerer token
+      if (!formID) return res.status(400).json({ error: "Bad Request" });
+      //Check an empty object or string
+      const expectedMaxRecords = parseInt(!maxRecords ? "10" : (maxRecords as string));
+      //Range is 1- 10
+      if (expectedMaxRecords < 1 || expectedMaxRecords > 10) {
+        return res.status(400).json({ error: "Invalid paylaod value found maxRecords" });
+      }
+      //Get token from request
+      const token = getBearerToken(req);
+      //Verify the token
+      const bearerTokenPayload = jwt.verify(
+        token,
+        process.env.TOKEN_SECRET || ""
+      ) as BearerTokenPayload;
+      const { email } = bearerTokenPayload;
+      //Check if a formUserRecord exists for the given bearerToken.
+      const formUserRecord = (
+        await getFormUserRecordByFormIDAndEmail(formID as string, email as string)
+      ).rows[0] as FormUserDBConfigProperties;
+      if (formUserRecord?.active) {
```

Comment:
If a record is returned `formUserRecord?.active` will always be true because of the SQL query on line 168 `"SELECT active FROM form_users WHERE template_id = ($1) and email = ($2) and active = true"`.  There might be a way to refactor this so that formUserRecord is checked to see if a result is returned instead.  This would also enable the FormUserDBConfigProperties to not have each property as optional but instead required.

Reply:
@bryan-robitaille  
Yeah the returned will always be active. I tried to avoid returning the hole row (unnecessary data)  but just  enough data.
Now i'm leaning towards this query  `SELECT EXISTS(SELECT 1 FROM form_users WHERE template_id = ($1) and email = ($2) and active = true)` which will return true or false so we can get rid of the entire formUSerRecords type.
OR 
`"SELECT  *  FROM form_users WHERE template_id = ($1) and email = ($2) and active = true"`  and make all formUserRecord's properties required. Then check if it is defined. 

What is your take on it ?

Reply:
I like the `SELECT EXISTS` and I see you've implemented it 👍